packages/cli/src/lib/config.ts

```diff
@@ -90,13 +90,20 @@ async function renameConfigIfNeeded(log: Log): Promise<void> {
   }
 }
 
-export async function loadOrCreateConfig(log: Log = new ConsoleLog('config'),
-    prompt: Prompt = new InquirerPrompt(), path = DEFAULT_CONFIG_FILE_PATH): Promise<Config> {
-  await renameConfigIfNeeded(log);
-  const existingConfig = await Config.loadConfig(path);
+export async function loadOrCreateConfig(
```

Comment:
I'm not sure I understand this change - it seems the end result is the same as the previous version.

In the section below, `path` is only assigned  `DEFAULT_CONFIG_FILE_PATH` if the value passed is different from `null` and different from `undefined`:

```ts
export async function loadOrCreateConfig(
    prompt: Prompt = new InquirerPrompt(), path = DEFAULT_CONFIG_FILE_PATH): Promise<Config> {
```
so, generally analog to:

```ts
  if (path === undefined) {
    await renameConfigIfNeeded(log);
    configPath = DEFAULT_CONFIG_FILE_PATH;
  } else {
    configPath = path;
  }
```

the only difference being the call to `await renameConfigIfNeeded(log)`. But that call should be no-op in most cases (it's just about handling old config paths to the current one).

Reply:
Well it _is_ calling out to filesystem which is a slow and unnecessary operation if we are already sure we don't need to do it, right?

Reply:
Not to mention you could be renaming a users input file which would be unexpected when rerunning a command.

Reply:
Hitting the filesystem shouldn't be such a large overhead in this case and renaming will only happen for a very specific filename, so I'm generally happy with the solution. Nevertheless, those are good points.

It sounds to me `loadOrCreateConfig()` is doing too much. Maybe a better alternative is to rename this function to `loadOrCreateDefaultConfig()` and when `parsedArgs.config` is available just call [`Config.loadConfig()`](https://github.com/GoogleChromeLabs/bubblewrap/blob/master/packages/core/src/lib/Config.ts#L43-L52) instead of this function.

PS: It's also possible to call `Config.deserialize()` to parse a JSON or build a config with a `new Config(jdkPath, androidSdkPath`). Would passing those strings from the command line make it a better solution for your use-case?

Reply:
I don't want to be in charge of function naming.

I think this is a very fast solution that can work for a lot of odd use case as anyone can make a config on the fly but they can also check in and use a local config file (mine is at `project/workspace/.bubblewrap/config.json`). This mean a user could say put the config in `package.json` and use `jq` to read and pass in through process substitution.

I plan to use Dhall everywhere and `dhall-to-json` for my configuration because I want types for my configuration (makes me want my `twa-manifest.json` to be typed too [which I'm working on](https://gitlab.com/toastal/dhall-webmanifest/-/blob/twa/TWAManifest/TrustedWebActivityManifest.dhall)). Also with this solution I can to call `nix-build --no-out-link '<nixpkgs>' -A jdk8_headless` to get the current version of JDK (same for Android SDK) on the fly which currently is at `/nix/store/q839v7q4k8s958zs2smmwm6id72xjsva-openjdk-headless-8u272-b10`. I need reproducible builds ... including all building tools and libraries, so everything needs to be local to `nix-shell` which includes `bubblewrap` and all its dependencies. I may make a derivation if time allows that contains all the bits that make this work on Nix.